planner/mapper/mapper.go

```diff
@@ -119,36 +120,91 @@ func resolveOrderDependencies(
 
 	// If there is orderby, and any one of the condition fields that are join fields and have not been
 	// requested, we need to map them here.
+outer:
 	for _, condition := range source.Value().Conditions {
-		if len(condition.Fields) <= 1 {
-			continue
-		}
-
-		joinField := condition.Fields[0]
+		fields := condition.Fields[:] // copy slice
```

Comment:
todo: this copy appears to be pointless, as the only other write is a full overwrite of the variable. You should be able to just do `fields := condition.Fields` here (as it was before)

Reply:
This actually smells a bit like a bug - do you mean to use `fields` for something besides a counter?  If so can you please add a failing test, then fix the bug (within this PR)?  Or otherwise quite heavily document why the counter is as such, yet you are not really using it for anything besides the `if`s

Reply:
Its copied because of this line further down in the `else if numFields > 2`:
```go
fields = fields[1:] // chop off the front item, and loop again on inner
```

Without the copy this would mutate the original slice.

Reply:
This is part of my point. You are copying the slice multiple times but only ever using its length.

Reply:
sorry, its *mutated* in place after its copied

Reply:
Apologies, talking with fred, I realized there is a bug (not sure why the tests arent tracking), I think this was potentially the result of a bad rebase, not sure. Currently its `joinField := condition.Fields[0]`, but it should be `joinField := fields[0]` which is why the `fields` slice keeps getting popped from the front.

It *can* still be replaced with length and index tracking, but the idea that the `fields` *should* be used to track the "recursion" depth thru the slice, and we just pop from the front. Will update and we can decide if it should be changed.

Reply:
Updated @AndrewSisley, see latest commit [here](https://github.com/sourcenetwork/defradb/pull/1500/commits/1c0d009b2d564dabafb575d627ec4efdaf23dffc). 

Hopefully makes more sense, and correcting the bug also affected another comment you had about the order stuff in the scanNode.

Reply:
Ah okay, makes sense now - thanks.  Any chance you can please add a test that would have caught that bug?  If it caught you out it can easily catch someone else out modifying the code in the future.

Reply:
Ill see if I can add one, but I don't think so beyond what we already have. The reason being, despite there being a bug here, it was accidentally being fixed in the scanNode `initCollection` method when it did its own `order` processing. When that was removed and this was fixed, they cancelled eachother out.

All 3 cases of this `if` block are covered by the existing tests.

Reply:
This makes a lot more sense now. Thanks for fixing!

Reply:
> All 3 cases of this if block are covered by the existing tests.

The lines are executed, but not necessarily tested.  Does reverting back to `joinField := condition.Fields[0]` cause the existing tests to fail?  If so, all good here :)